src/include/wiredtiger.in

```diff
@@ -39,6 +39,16 @@ extern "C" {
 #define	__F(func) (*(func))
 #endif
 
+/*
+ * We support configuring WiredTiger with the gcc/clang -fvisibility=hidden
+ * flags, but that requires public APIs be specifically marked.
+ */
+#if defined(DOXYGEN) || defined(SWIG) || !defined(__GNUC__)
+#define	__LIBRARY_VISIBLE
+#else
+#define	__LIBRARY_VISIBLE	__attribute__ ((visibility("default")))
+#endif
+
```

Comment:
Why not use WT_LIBRARY_VISIBLE here (and not need the internal versions in `gcc.h`, `lint.h` and `msvc.h`) instead of another macro starting with underscores?

We should consider defining the Windows version to `__declspec(dllexport)` and get rid of `build_win/wiredtiger.def`.